### ISSUE TITLE
DateTimeUpDown FormatString causes crash on japanese system

### ISSUE THREAD

**Issue by Schnjan** [xceedsoftware/wpftoolkit#1191]
**Description**

When using the japanese culture with the japanese calendar, DateTimeUpDown and its derivates throw a XamlParseException. This is due to the validation check, which simply parses DateTime.MinValue with the specified format. On the japanese calendar (and others) this results in an ArgumentOutOfRangeException, because it starts somewhere in the 1800s.

**Steps to reproduce**

- Create a new WPF application
- Create a DateTimeUpDown in MainWindow.xaml
- Set the FormatString to 'G'
- Add the following lines to the 'OnStartUp' method in App.xaml.cs
```
CultureInfo info = new CultureInfo("ja-JP");
info.DateTimeFormat.Calendar = new JapaneseCalendar();
Thread.CurrentThread.CurrentCulture = info;
```

- Start the application

**Actual behavior**

Crash

**Excpected behavior**

No crash

**Comment by XceedBoucherS** [comment]
Hi,

When trying to set DatetimeUpDown.FormatString to "G", the application crashes because :
{"'G' is not a valid value for property 'FormatString'."}
When removing the FormatString, the application starts.

1)  When the application starts, clicking the Up arrow will display the current dateTime. When clicking again the up arrow, application crashes in DateTimeUpDown.Increment() because the newValue is null and we are trying to obtain its Value. This is already fixed in v3.5.

2) With v3.5, selecting a digit value in the DateTimeUpDown and pressing the Up/down arrows will correctly increment/decrement the value.

3) How do you get the ArgumentOutOfRangeException ?

**Comment by Schnjan** [comment]
The ArgumentOutOfRangeException is catched by the DateTimeUpDown in the validator of the FormatString dependency property. The validator then returns false, which leads to the WPF-generated message ('G' is not a valid foo...). 

1. 'G' is a valid format string, and a very common one. I don't want to remove it
2. When using 'G' with a default european, american or even japanese culture, it works fine
3. The crash occurs directly when the XAML is loaded, and only with the japanese calendar, like shown in the code sample above

Please check **line 91** in **DateTimeUpDown.cs**

**Comment by XceedBoucherS** [comment]
Hi,

Thanks for the details. You are right. 

The solution consist in replacing the line :
DateTime.MinValue.ToString( ( string )value, CultureInfo.CurrentCulture );
with :
CultureInfo.CurrentCulture.DateTimeFormat.Calendar.MinSupportedDateTime.ToString( (string)value, CultureInfo.CurrentCulture );
in the "DateTimeUpDown.IsFormatStringValid" method.

This will be fixed in v3.6.
All other affected controls will also be fixed in v3.6.

**Comment by Schnjan** [comment]
Thank you!

**Comment by XceedBoucherS** [comment]
Actually, v3.2 is already released. So the fix will be in v3.6 free nuget version. when that version will be released.
